### PR TITLE
remove uses of type() and type_as() part 1.

### PR DESCRIPTION
**Pull request by ailzhang**

### CODE REVIEW

**Review by mrshenli — APPROVED** [review]

File: test/distributed/test_distributed.py [hunk 1]
```diff
@@ -655,24 +655,24 @@ def _test_broadcast_helper(
         self, group, group_id, rank, cuda=False, rank_to_GPU=None
     ):
         for ttype, value, requires_cuda in [
```
Comment (SsnL) [inline test/distributed/test_distributed.py:657]: probably should rename ttype->dtype

File: test/test_nn.py [hunk 2]
```diff
@@ -7469,7 +7473,7 @@ def test_upsamplingBilinear2d(self):
 
     def test_upsamplingBicubic2d(self):
         # test output against known input: align_corners=False result must match opencv
-        in_t = torch.arange(8).view(1, 2, 2, 2).type(torch.FloatTensor)
+        in_t = torch.arange(8).view(1, 2, 2, 2).to(torch.float)
```
Comment (gchanan) [inline test/test_nn.py:7476]: you can probably just change this (and the one below) to `torch.arange(8.)`

File: test/distributed/test_distributed.py [hunk 3]
```diff
@@ -654,25 +654,25 @@ def test_irecv(self):
...
+            (torch.int8, -2, False),
+            (torch.uint8, 129, False),
+            (torch.int, -1e5, False),
+            (torch.long, -1e15, False),
         ]:
             if requires_cuda and not cuda:
                 continue
             for src in group:
-                expected_tensor = _build_tensor(src + 1, value).type(ttype)
+                expected_tensor = _build_tensor(src + 1, value).to(dtype=dtype)
```
Comment (ngimel) [inline test/distributed/test_distributed.py:669]: maybe change `_build_tensor` to accept dtype? Currently it's hardcoded to float and uses old-style api.

File: test/test_jit.py [hunk 4]
```diff
@@ -6674,7 +6674,7 @@ def test_integral_shape_inference(self):
         def test_integral_shape_inference(a):
             return a / a
         ''')
-        inputs = [torch.ones(10, 10).type(torch.LongTensor)]
+        inputs = [torch.ones(10, 10).to(torch.long)]
```
Comment (ngimel) [inline test/test_jit.py:6677]: torch.ones(10,10, dtype=torch.long)

File: test/test_multiprocessing.py [hunk 5]
```diff
@@ -48,23 +48,23 @@ def simple_pool_fill(tensor):
     return tensor.add(1)
 
 
-def send_tensor(queue, event, tp):
-    t = torch.ones(5, 5).type(tp)
+def send_tensor(queue, event, device, dtype):
+    t = torch.ones(5, 5).to(device, dtype)
```
Comment (ngimel) [inline test/test_multiprocessing.py:52]: torch.ones(5,5, device=device, dtype=dtype), same in other places.

File: test/test_multiprocessing.py [hunk 6]
```diff
@@ -48,23 +48,23 @@ def simple_pool_fill(tensor):
...
     queue.put(t)
     queue.put(t)
     event.wait()
 
 
-def send_and_delete_tensors(queue, event, tp, count, size=5):
+def send_and_delete_tensors(queue, event, device, dtype, count, size=5):
     for i in range(count):
-        t = torch.full([size], i).type(tp)
+        t = torch.full([size], i).to(device, dtype)
```
Comment (ngimel) [inline test/test_multiprocessing.py:60]: same, use device and dtype kwargs in the factory functions

File: test/test_nn.py [hunk 7]
```diff
@@ -6193,15 +6193,17 @@ def test_batchnorm_cudnn_half(self):
         thnn_output = m(input)
         thnn_output.sum().backward()
         thnn_input_grad = input.grad.data.clone()
-        self.assertEqual(thnn_output.type(), input.type())
+        self.assertEqual(thnn_output.device, input.device)
```
Comment (ngimel) [inline test/test_nn.py:6196]: These 2 lines are encountered sufficient number of times to merit their own helper function?

File: test/test_nn.py [hunk 8]
```diff
@@ -10029,15 +10037,14 @@ def _padded_sequence(self, tensor_type):
...
-            cuda_type_str = 'torch.cuda.FloatTensor'
             padded, lengths = self._padded_sequence(tensor_type)
             packed = rnn_utils.pack_padded_sequence(
                 padded, lengths, enforce_sorted=enforce_sorted)
             self.assertFalse(packed.is_cuda)
             packed = packed.to(device)
             self.assertTrue(packed.is_cuda)
             unpacked, _ = rnn_utils.pad_packed_sequence(packed)
-            self.assertEqual(unpacked.type(), cuda_type_str)
+            self.assertEqual(unpacked.dtype, torch.float)
```
Comment (ngimel) [inline test/test_nn.py:10047]: Here you are not testing that unpacked in on cuda (it was tested before)

File: test/test_nn.py [hunk 9]
```diff
@@ -10029,15 +10037,14 @@ def _padded_sequence(self, tensor_type):
     def test_device_mask(self, device):
         for enforce_sorted in [True, False]:
             tensor_type = torch.FloatTensor
```
Comment (ngimel) [inline test/test_nn.py:10039]: maybe get rid of tensor_type to, and use .to(torch.float)?

**Review by ngimel — APPROVED** [review]

**Comment by mruberry** [comment]
Part of addressing https://github.com/pytorch/pytorch/issues/37863.

**Pull request closed without merging**